pkg/icingaredis/v1/stats_message.go

```diff
@@ -24,25 +25,25 @@ func (m StatsMessage) IcingaStatus() (*IcingaStatus, error) {
 		}
 
 		if err := json.Unmarshal([]byte(s), &envelope); err != nil {
-			return nil, errors.Wrap(err, "can't parse Icinga 2 status")
+			return nil, internal.CantUnmarshalJSON(err, []byte(s), envelope)
 		}
 
 		return &envelope.Status.IcingaApplication.IcingaStatus, nil
 	}
 
-	return nil, errors.New("bad message")
+	return nil, errors.Errorf("bad message %#v. %s missing", m, "IcingaApplication")
```

Comment:
Why not just inline the hardcoded string constant parameter into the format string?